nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.core;
+
+import static javax.servlet.http.HttpServletResponse.SC_CONFLICT;
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.*;
+import org.nuxeo.ecm.core.blob.ColdStorageHelper;
+import org.nuxeo.ecm.core.schema.FacetNames;
+import org.nuxeo.ecm.core.test.ColdStorageFeature;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(ColdStorageFeature.class)
+public class TestColdStorage {
+
+    protected static final String FILE_CONTENT = "foo";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Test
+    public void shouldMoveBlobDocumentToColdStorage() throws IOException {
+        DocumentModel documentModel = createDocument(true);
+
+        // move the blob to cold storage
+        documentModel = ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+        transactionalFeature.nextTransaction();
+        documentModel.refresh();
+
+        assertTrue(documentModel.hasFacet(FacetNames.COLD_STORAGE));
+
+        assertNull(documentModel.getPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY));
+
+        // check if the `coldstorage:coldContent` contains the original file content
+        checkBlobContent(documentModel, ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY, FILE_CONTENT);
+    }
+
+    @Test
+    public void shouldFailWhenMovingDocumentBlobAlreadyInColdStorage() {
+        DocumentModel documentModel = createDocument(true);
+
+        // move for the first time
+        documentModel = ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+
+        // try to make another move
+        try {
+            ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+            fail("Should fail because the content is already in cold storage");
+        } catch (NuxeoException ne) {
```

Comment:
Juste call the exception the traditional `e` if there's no chance of confusion (in several methods).